### PR TITLE
Ab#111548 authenticate workload management

### PR DESCRIPTION
**Pull request by AlexandreSanscartier**

### CODE REVIEW

File: src/CSF.SRDashboard.Client/Graph/GraphConstants.cs [hunk 1]
```diff
@@ -5,7 +5,13 @@ public static class GraphConstants
         // Defines the permission scopes used by the app
         public readonly static string[] Scopes =
         {
-            "User.Read"
+            "User.Read",
+        };
+
+        public readonly static string[] WorkloadManagementAPIScopes =
+        {
+            "https://034gc.onmicrosoft.com/ncd-wms-dev/WorkItem.Read.All",
```
Comment (tc-chris) [inline src/CSF.SRDashboard.Client/Graph/GraphConstants.cs:13]: Shouldn't these be in appsettings.json?

File: src/CSF.SRDashboard.Client/appsettings.json [hunk 2]
```diff
@@ -9,23 +9,24 @@
...
 
-
   "AzureGraphAPI": {
     "UserProfileURL": "https://graph.microsoft.com/beta/me",
     "UserPhotoURL": "https://graph.microsoft.com/v1.0/me/photo/$value"
   },
 
+  "WorkloadManagementAPI": {
+    "BaseUrl": "https://ncdsafsecsurapp-work-management-service.azurewebsites.net/",
+    "Scopes": "https://034gc.onmicrosoft.com/ncd-wms-dev/WorkItem.Read.All https://034gc.onmicrosoft.com/ncd-wms-dev/WorkItem.CreateUpdate.All"
```
Comment (tc-chris) [inline src/CSF.SRDashboard.Client/appsettings.json:19]: Why do we have them in one line and not in an array?

**Review by tc-chris — CHANGES_REQUESTED** [review]

File: src/CSF.SRDashboard.Client/appsettings.json (hunk 2 above)
Comment (AlexandreSanscartier) [reply src/CSF.SRDashboard.Client/appsettings.json:19]: This is how the DownstreamWebAPI works. You need them specifically like that in order for it to work.

File: src/CSF.SRDashboard.Client/Graph/GraphConstants.cs (hunk 1 above)
Comment (AlexandreSanscartier) [reply src/CSF.SRDashboard.Client/Graph/GraphConstants.cs:13]: Will remove them, good catch I believe we don't need them in there.

File: src/CSF.SRDashboard.Client/appsettings.json (hunk 2 above)
Comment (tc-chris) [reply src/CSF.SRDashboard.Client/appsettings.json:19]: Okay

File: src/CSF.SRDashboard.Client/Pages/ViewRequestDetails.razor.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ protected async override Task OnInitializedAsync()
 
             this.Applicant = this.GatewayService.GetApplicantInfoByCdn(Cdn);
 
-            WorkItemDTO = this.WorkLoadService.GetByWorkItemById(RequestId);
+            WorkItemDTO = await this.WorkLoadService.GetByWorkItemById(RequestId);
```
Comment (TC-Cory) [inline src/CSF.SRDashboard.Client/Pages/ViewRequestDetails.razor.cs:62]: Not sure if you've received this error as well, I am getting a null reference on the RequestModel in AssigneeSelector.razor.cs. I've worked around this by initializing RequestModel before loading the workItem.

**Pull request merged**